docs/contribution/nearcore.md

```diff
@@ -114,3 +114,108 @@ Official image is published at `nearprotocol/nearcore`
 <a href="https://stackoverflow.com/questions/tagged/nearprotocol">
   <h8> Ask it on stack overflow! </h8>
 </a>
+
+### How to speed up rust compilation time
+
+This describes on how to improve nearcore compilation time without having to
+modify Cargo.toml. It's possible to override Cargo.toml setting by setting
+`RUSTFLAGS` variable and also by adding a compilation cache by overriding
+`RUSTC_WRAPPER`.
+
+
+# Default build
+```bash
+# time RUSTFLAGS= RUSTC_WRAPPER= cargo build -p neard --release
+real	8m59.257s user	40m11.663s sys	2m8.809s
+# touch */*/*/*.rs; time RUSTFLAGS= RUSTC_WRAPPER= cargo build -p neard --release
+real	5m44.904s user	15m53.672s sys	0m16.913s
+```
+
+# Turn off linking time optimization
+May have low performance impact.
+```bash
+# time RUSTFLAGS='-C lto=off' RUSTC_WRAPPER= cargo build -p neard --release
+real	6m25.495s user	40m54.353s sys	2m43.932s
+# touch */*/*/*.rs; time RUSTFLAGS='-C lto=off' RUSTC_WRAPPER= cargo build -p neard --release
+real	1m38.721s user	7m45.446s sys	0m11.767s
+```
+
+# Use LLD linker
+```bash
+# time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld' RUSTC_WRAPPER= cargo build -p neard --release
+real	5m57.307s user	37m28.079s sys	2m27.664s
+# touch */*/*/*.rs; time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld' RUSTC_WRAPPER= cargo build -p neard --release
+real	1m26.627s user	7m12.705s sys	0m7.900s
+
+```
+
+# experimental share-generics feature
+```bash
+# time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld -Zshare-generics=y' RUSTC_WRAPPER= cargo build -p neard --release
+real	4m35.538s user	29m17.550s sys	2m6.262s
+# touch */*/*/*.rs; time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld -Zshare-generics=y' RUSTC_WRAPPER= cargo build -p neard --release
+real	0m53.101s user	3m52.850s sys	0m7.317s
+```
+
+# parallelize building invidual cargo packages
+```bash
+# time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld -Zshare-generics=y -C codegen-units=6' RUSTC_WRAPPER= cargo build -p neard --release
+real	4m42.485s user	28m47.223s sys	2m18.833s
+# touch */*/*/*.rs; time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld -Zshare-generics=y -C codegen-units=6' RUSTC_WRAPPER= cargo build -p neard --release
+real	0m51.620s user	4m13.180s sys	0m11.254s
+```
+
+# cache results from previous compilations using sccache
+```bash
+# time RUSTFLAGS='-C lto=off -C link-arg=-fuse-ld=lld -Zshare-generics=y -C codegen-units=6' RUSTC_WRAPPER=sccache cargo build -p neard --release
```

Comment:
this one seems to be exactly the same as the command below

Reply:
how is this resolved?

Reply:
It's fixed now..